config/src/main/java/com/quorum/tessera/config/constraints/ServerConfigsValidator.java

```diff
@@ -13,40 +14,25 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ServerConfigsValidator.class);
 
     @Override
-    public boolean isValid(List<ServerConfig> serverConfigs, ConstraintValidatorContext constraintContext) {
-        if(serverConfigs == null) {
+    public boolean isValid(final List<ServerConfig> serverConfigs, final ConstraintValidatorContext constraintContext) {
+        if (serverConfigs == null) {
             return true;
         }
-        
-        int p2PEnabledConfigsCount = 0;
-        int q2TEnabledConfigsCount = 0;
-
-        for (ServerConfig sc : serverConfigs) {
-            if (sc.isEnabled()) {
-                switch (sc.getApp()) {
-                    case Q2T:
-                        q2TEnabledConfigsCount++;
-                        break;
-                    case P2P:
-                        p2PEnabledConfigsCount++;
-                        break;
-                }
-            }
-        }
+
+        final long p2PEnabledConfigsCount = serverConfigs
+            .stream()
+            .filter(ServerConfig::isEnabled)
+            .filter(sc -> AppType.P2P.equals(sc.getApp()))
+            .count();
 
         if (p2PEnabledConfigsCount != 1) {
-            LOGGER.debug("Only one P2P server must be configured and enabled.");
-            constraintContext.disableDefaultConstraintViolation();
-            constraintContext.buildConstraintViolationWithTemplate("Only one P2P server must be configured and enabled.")
-                .addConstraintViolation();
-            return false;
-        }
+            LOGGER.warn("Exactly one P2P server must be configured and enabled.");
```

Comment:
Considering people are having  multiple quorum nodes connected to a single tessera node - there may be scenarios where even the P2P is not required... :)

Reply:
This is true, but I think it is a little more complicated than that. There are other parts of the code where we make use of P2P API's (such as key validation) and I did not look deeply into what is exactly affected.

I agree that eventually we should relax this restriction, but it can be done separately. The Q2T one was an easy win for now.